harmonica/equivalent_layer/harmonic.py

```diff
@@ -0,0 +1,228 @@
+        a single point source bellow each obervation point at a depth given by the
+        product of the ``depth_factor`` and the mean distance to the nearest
+        k obervation points. A greater ``depth_factor`` will increase the depth of the
+        point source. This parameter is ignored if ``points`` is not None. Default 3.
+    k_nearest : int
+        Number of observation points used to compute the median distance to its nearest
+        neighbours. This argument is passed to :func:`verde.mean_distance`. It's ignored
+        if ``points`` is not None. Default 1.
+
+    Attributes
+    ----------
+    points_ : 2d-array
+        Coordinates of the point sources used to build the Equivalent Layer.
+    coefs_ : array
+        Estimated coefficients of every point source.
+    region_ : tuple
+        The boundaries (``[W, E, S, N]``) of the data used to fit the interpolator.
+        Used as the default region for the :meth:`~harmonica.HarmonicEQL.grid` and
+        :meth:`~harmonica.HarmonicEQL.scatter` methods.
+    """
+
+    def __init__(self, damping=None, points=None, depth_factor=3, k_nearest=1):
+        self.damping = damping
+        self.points = points
+        self.depth_factor = depth_factor
+        self.k_nearest = k_nearest
+
+    def fit(self, coordinates, data, weights=None):
+        """
+        Fit the coefficients of the Equivalent Layer.
+
+        The data region is captured and used as default for the
+        :meth:`~harmonica.HarmonicEQL.grid` and :meth:`~harmonica.HarmonicEQL.scatter`
+        methods.
+
+        All input arrays must have the same shape.
+
+        Parameters
+        ----------
+        coordinates : tuple of arrays
+            Arrays with the coordinates of each data point. Should be in the
+            following order: (easting, northing, upward, ...). Only easting
+            and northing will be used, all subsequent coordinates will be
+            ignored.
+        data : array
+            The data values of each data point.
+        weights : None or array
+            If not None, then the weights assigned to each data point.
+            Typically, this should be 1 over the data uncertainty squared.
+
+        Returns
+        -------
+        self
+            Returns this estimator instance for chaining operations.
+        """
+        coordinates, data, weights = check_fit_input(coordinates, data, weights)
+        # Capture the data region to use as a default when gridding.
+        self.region_ = get_region(coordinates[:2])
+        coordinates = n_1d_arrays(coordinates, 3)
+        if self.points is None:
+            # Put a single point source bellow each observation point at a depth three
+            # times the meadian distance to the nearest k observation points.
+            point_east, point_north, point_upward = tuple(i.copy() for i in coordinates)
+            point_upward += self.depth_factor * median_distance(
```

Comment:
After we merge #92, the plus sign must be a minus in order to actually put the point sources below the observation points.
```suggestion
            point_upward -= self.depth_factor * median_distance(
```